common/database/connector.php

```diff
@@ -0,0 +1,114 @@
+ *
+ * If needed, the instance can be used in a transactional context:
+ * <code>
+ *     $database = \common\database\Connector::Instance();
+ *     $entityManager = $database->entityManager;
+ *     $entityManager->getConnection()->beginTransaction();
+ *     try {
+ *         // Entity work here...
+ *         $entityManager->persist($someEntityToBePersisted);
+ *         $entityManager->flush();
+ *         $entityManager->getConnection()->commit();
+ *     } catch (Exception $e) {
+ *         $entityManager->getConnection()->rollBack();
+ *         throw $e;
+ *     }
+ * </code>
+ *
+ * Copyright (C) 2016 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+namespace common\database;
+
+use \Doctrine\ORM\Tools\Setup;
+use \Doctrine\ORM\EntityManager;
+use \common\database\Auditor;
+
+final class Connector {
+    /**
+     * The pooled Doctrine connection.
+     */
+    public $entityManager;
+
+    /**
+     * Logger for noting connection/configuration information.
+     */
+    private $logger;
+
+    /**
+     * Default constructor.
+     */
+    private function __construct() {
+        $this->logger = new \common\logging\Logger("\common\database\Connector");
+        $this->createConnection();
+    }
+
+    /**
+     * The only public method for consumers to either create or gain access to the sole singleton
+     * instance of the class.
+     *
+     * @return Connector instance
+     */
+    public static function Instance() {
+        static $singletonInstance = null;
+        if ($singletonInstance === null) {
+            $singletonInstance = new Connector();
+        }
+
+        return $singletonInstance;
+    }
+
+    /**
+     * Creates the pooled Doctrine connection. All connection data is configurable via sqlconf.php.
+     * By default, the connection is pooled, in a nondev mode, and uses the pdo_mysql driver. Note
+     * that $GLOBALS["doctrine_connection_pooling"] and $GLOBALS["doctrine_dev_mode"] are used instead
+     * of $sqlconf[] because editing the sqlconf.php is not allowed (will mess up endusers trying to
+     * upgrade their install).
+     *
+     * @todo document throwables
+     */
+    private function createConnection() {
```

Comment:
The following 2 things need to be built into the sql connection:
1. UTF8 (dependent on flag)
https://github.com/openemr/openemr/blob/master/library/sql.inc#L131-L137

2. clear sql_mode
https://github.com/openemr/openemr/blob/master/library/sql.inc#L139-L143